### ISSUE TITLE
"Propagated" too far?

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2639]
Migrated from [rt.perl.org#4330](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=4330) (status was 'resolved')

Searchable as RT4330$

**Comment by p5pRT** [comment]
### From domo@computer.org

#### Created by domo@computer.org


This with perl@&#8203;7094&#8203;:

% \./miniperl \-Ilib \-Mcharnames=icelandic \-le \\
'$s="\\N\{LATIN SMALL LETTER THORN\}"; \{use bytes; print length $s\, $s\}'
Unknown charname 'LATIN SMALL LETTER THORN' at lib/charnames\.pm line 31\.
Propagated at \-e line 1\, within string
Unknown charname 'LATIN SMALL LETTER THORN' at lib/charnames\.pm line 31\.
PropagatedBEGIN not safe after errors\-\-compilation aborted at \-e line 1\.

\(Apologies for the long on\-liner\.\)

There seem to be several problems here&#8203;:

1\. The second "Propagated" looks to me as if it has leaked out into a
&nbsp;    message where it's not required\.

2\. If I'm wrong\, and it is required on the last line\, its formatting
&nbsp;    is incorrect \-\- there's a missing trailing space\.

3\. "Propagated" an error message doesn't make much sense to me\, and
&nbsp;    it's not documented in perldiag\.pod\.

4\.  charnames should probably whine if I specify a stupid script name\,
&nbsp;     but that's a separate issue\.

The "Propagated" comes from toke\.c line 5762\, but as I don't know what
should be happening\, I can't attempt to fix things\.

Enlightenment\, please\.

<details><summary>Perl Info</summary>

```

Flags:
     category=core
     severity=medium

This perlbug was built using Perl v5.7.0 - Sat Sep 16 15:03:48 WET DST 2000
It is being executed now by  Perl v5.7.0 - Tue Sep  5 16:08:31 WET DST 2000.

Site configuration information for perl v5.7.0:

Configured by domo at Tue Sep  5 16:08:31 WET DST 2000.

Summary of my perl5 (revision 5.0 version 7 subversion 0) configuration:
   Platform:
     osname=machten, osvers=4.1.4, archname=powerpc-machten
     uname='machten ppp100 5 0.5 powerpc '
     config_args='-Doptimize=-g -e'
     hint=recommended, useposix=true, d_sigaction=define
     usethreads=undef use5005threads=undef useithreads=undef 
usemultiplicity=undef
     useperlio=undef d_sfio=undef uselargefiles=define usesocks=undef
     use64bitint=undef use64bitall=undef uselongdouble=undef
   Compiler:
     cc='cc', ccflags ='-DNOTDEF_MACHTEN -DREG_INFTY=2047 -DDEBUGGING 
-I/usr/local/include', optimize='-g', cppflags='-DNOTDEF_MACHTEN 
-DREG_INFTY=2047 -DDEBUGGING -I/usr/local/include'
     ccversion='', gccversion='2.8.1', gccosandvers='machten4'
     intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=4321
     d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=8
     ivtype='long', ivsize=4, nvtype='double', nvsize=8, 
Off_t='off_t', lseeksize=4
     alignbytes=8, usemymalloc=y, prototype=define
   Linker and Libraries:
     ld='ld', ldflags =' -Xlstack=1048576 -L/usr/local/lib'
     libpth=/usr/local/lib /usr/lib
     libs=-lndbm -lgdbm -ldb -lm -lc -lutil
     libc=/usr/lib/libc.a, so=so, useshrplib=false, libperl=libperl.a
   Dynamic Linking:
     dlsrc=dl_none.xs, dlext=none, d_dlsymun=undef, ccdlflags=''
     cccdlflags='', lddlflags=''

Locally applied patches:
     DEVEL7093


@INC for perl v5.7.0:
     /usr/local/lib/perl5/5.7.0/powerpc-machten
     /usr/local/lib/perl5/5.7.0
     /usr/local/lib/perl5/site_perl/5.7.0/powerpc-machten
     /usr/local/lib/perl5/site_perl/5.7.0
     /usr/local/lib/perl5/site_perl/5.6.0/powerpc-machten
     /usr/local/lib/perl5/site_perl/5.6.0
     /usr/local/lib/perl5/site_perl
     .


Environment for perl v5.7.0:
     HOME=/home/domo
     LANG (unset)
     LANGUAGE (unset)
     LD_LIBRARY_PATH=/usr/lib
     LOGDIR (unset)
 
PATH=/sbin:/usr/sbin:/home/domo/bin:/usr/local/bin:/usr/bin:/bin:/usr/X11/bin:/usr/libexec
     PERL_BADLANG (unset)
     SHELL=/bin/bash


```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

This ticket can be closed&#8203;: patch 7178 fixed it \-\- for a while&#8203;: 
problems in this area have resurfaced\, so I'll submit a new bug 
report\.
